aiida_registry/fetch_metadata.py

```diff
@@ -192,6 +229,16 @@ def complete_plugin_data(
                     plugin_data["aiida_version"] = data.aiida_version
                     plugin_data["entry_points"] = data.entry_points
 
+    current_version = get_last_fetched_version(plugin_data["name"])
+    if current_version:
+        try:
+            if current_version != plugin_data["metadata"]["version"]:
```

Comment:
Just for me to understand: when would these versions be different?
Don't both of them come from the same source (the `plugins_metadata.json` file)?

If we are anyhow using PyPI for the last release date, is this logic still necessary?

Reply:
The `get_last_fetched_version()` function gets the version from the last committed file in the repository, while `plugin_data["metadata"]["version"]` is the version of the new `plugins_metadata.json` file to be constructed.
After applying this, every plugin that has a change of version will have a release date, not only plugins on PyPI.


I think what I tried to do here is not clear because this depends on merging things on `master`. because the job of commiting `plugins_metadata.json` is on `workflow.yml`. https://github.com/aiidateam/aiida-registry/blob/05f8aa444cb662d00212ec5eb76b77452e797c10/.github/workflows/webpage.yml#L40C1-L47C39

Reply:
Right, so the plugins that are not on PyPI will have no release date in the `plugins_metadata.json` until the day when they make a new release.
At that point, a release date will be added here.

Is that correct?

Reply:
Yes, exactly!